plugins/follow_me/follow_me.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include <iostream>
+#include <cmath>
+
+namespace dronecore {
+
+class FollowMeImpl;
+
+class FollowMe
+{
+public:
+    explicit FollowMe(FollowMeImpl *impl);
+    ~FollowMe();
+
+    /**
+     * @brief Results of FollowMe operations
+     */
+    enum class Result {
+        SUCCESS = 0, /**< @brief %Request succeeded. */
+        NO_DEVICE, /**< @brief No device connected. */
+        CONNECTION_ERROR, /**< @brief %Connection error. */
+        BUSY, /**< @brief Vehicle busy. */
+        COMMAND_DENIED, /**< @brief Command denied. */
+        TIMEOUT, /**< @brief %Request timeout. */
+        UNKNOWN /**< @brief Unknown error. */
+    };
+
+    /**
+     * @brief Altitude flag for Follow Target command.
+     */
+    enum class ESTCapabilities {
+        POS = 0,
+        VEL = 1,
+        ACCEL = 2,
+        ATT_RATES = 3
+    };
+
+    /**
+     * @brief Motion report info which will be emitted peridically to the Vehicle
+     */
+    struct MotionReport {
+        // set to home position used by PX4.
+        int32_t lat_int = static_cast<uint32_t>(47.3977419 *
```

Comment:
1. A motion report should be as user-friendly as possible and therefore use `double` for lat/lon.
2. Also, it should not have any default values since the user needs to set them.
3. All variables need to be clearly named, no abbreviations, and always with units in the variable name.
4. The docs need to have `@brief` and end with `.`.

Reply:
Agree to points 1, 3 & 5.
But for the point 2, I prefer to have them start with default values initialized.
In real scenario, I believe  user don't provide position at all. I see that QGroundControl doesn't prompt for position in _FollowMe_ mode.
Aren't we exporting `set_position` API to for testing purpose ?
As per [PX4 Flight modes](https://docs.px4.io/en/flight_modes/) documentation, In Follow Me mode, the vehicle autonomously follows a user with an Android phone/tablet running QGroundControl.